businessRules/users.js

```diff
@@ -18,70 +25,38 @@ function registerUser(userName, password, firstName, lastName) {
 }
 
 function login(userName, password) {
-    if (!userName) {
-        throw new Error('Error. There was no username provided.');
-    }
-    if (!password) {
-        throw new Error('Error. There was no password provided.');
-    }
-
-    let users = getUsers();
+    validateUserName(userName);
 
-    const user = users.find(
-        (user) => user.userName === userName && user.password === password
-    );
+    validatePassword(password);
 
-    if (!user) {
-        throw new Error('Error. No such user.');
-    }
+    const user = findUserWithSameUserNameAndPassword(userName, password);
 
     const sessionId = md5(v4()); //Should be REAL cryptography.
+
     updateUser(user.id, 'session', sessionId);
 
     return sessionId;
 }
 
-function usersList() {
-    return getUsers();
-}
-
 function deleteUserSession(userId) {
     updateUser(userId, 'session', null);
 }
 
 function logout(userLoginSession) {
-    if (!userLoginSession) {
-        throw new Error('Error. There was no users session data provided.');
-    }
+    validateLoginSession(userLoginSession);
 
-    let users = getUsers();
-
-    const user = userLoginSession
-        ? users.find((user) => user.session === userLoginSession)
-        : null;
-
-    if (!user) {
-        throw new Error('You are not logged in.');
-    }
+    const user = findUserByUserLoginSession(userLoginSession);
 
     deleteUserSession(user.id);
 }
 
 function updateUser(userId, key, value) {
-    if (!key) {
-        throw new Error("Error. You didn't provide any key to update.");
-    }
-    if (value === undefined) {
-        throw new Error("Error. You didn't provide any value to update.");
-    }
+    validateKey(key);
 
-    let users = getUsers();
+    validateValue(value);
 
-    const user = users.find((user) => userId === user.id);
-
-    if (!user) {
-        throw new Error('Error. No such user.');
-    }
+    const users = getUsers(),
+        user = getUser(userId);
```

Comment:
Can you break it into two separate statements?

Reply:
Done.